plinth/modules/matrixsynapse/views.py

```diff
@@ -54,6 +60,38 @@ def get_context_data(self, *args, **kwargs):
 
         return context
 
+    def get_initial(self):
+        """Return the status of the service to fill in the form."""
+        return {
+            'is_user_registrations_enabled':
+            matrixsynapse.is_user_registrations_enabled()
+        }
+
+    def form_valid(self, form):
+        """Enable/disable user registrations"""
+        old_registration = form.initial['is_user_registrations_enabled']
+        new_registration = form.cleaned_data['is_user_registrations_enabled']
+        MATRIXSYNAPSE_YAML = "/etc/matrix-synapse/homeserver.yaml"
+
+        if old_registration == new_registration:
+            if not self.request._messages._queued_messages:
+                messages.info(self.request, _('Setting unchanged'))
+        else:
+            if new_registration:
+                matrixsynapse.enable_user_registrations()
+                messages.success(self.request, _('User registrations enabled'))
+
+                with YAMLFile(MATRIXSYNAPSE_YAML) as conf:
```

Comment:
Here you're changing the config file directly - would be much better to use the action you defined.